### ISSUE TITLE
Simplest use fails with error “Length cannot be less than zero.”

### ISSUE THREAD

**Issue by atifaziz** [loxsmoke/mddox#2]
## Steps to reproduce

Assume a PowerShell session.

I installed the [morelinq](https://www.nuget.org/packages/morelinq) package using:

    nuget install morelinq

The installation of the package succeeded with NuGet printing:

```
Feeds used:
  https://api.nuget.org/v3/index.json
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\

Installing package 'morelinq' to 'T:\'.
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/morelinq/index.json


Attempting to gather dependency information for package 'morelinq.3.2.0' with respect to project 'T:\', targeting 'Any,Version=v0.0'
Gathering dependency information took 43.8 ms
Attempting to resolve dependencies for package 'morelinq.3.2.0' with DependencyBehavior 'Lowest'
Resolving dependency information took 0 ms
Resolving actions to install package 'morelinq.3.2.0'
Resolved actions to install package 'morelinq.3.2.0'
Retrieving package 'morelinq 3.2.0' from 'nuget.org'.
Found package 'System.ValueTuple 4.4.0' in 'T:\'.
Package 'System.ValueTuple.4.4.0' already exists in folder 'T:\'
Successfully installed 'System.ValueTuple 4.4.0' to T:\
Adding package 'morelinq.3.2.0' to folder 'T:\'
Added package 'morelinq.3.2.0' to folder 'T:\'
Successfully installed 'morelinq 3.2.0' to T:\
Executing nuget actions took 204.68 ms
```

Then I listed the assemblies in the packages, using:

    dir -Recurse -File morelinq.3.2.0\*.dll | select -ExpandProperty FullName

The output was:

```
T:\morelinq.3.2.0\lib\net451\MoreLinq.dll
T:\morelinq.3.2.0\lib\netstandard1.0\MoreLinq.dll
T:\morelinq.3.2.0\lib\netstandard2.0\MoreLinq.dll
```

I then executed the following to generate documentation:

    mddox morelinq.3.2.0\lib\netstandard2.0\MoreLinq.dll

The tool failed with the following error message:

```
Error: Length cannot be less than zero.
Parameter name: length
   at System.String.Substring(Int32 startIndex, Int32 length)
   at DocXml.Reflection.ReflectionExtensions.ToNameString(Type type, Queue`1 tupleFieldNames, Func`3 typeNameConverter)
   at MdDox.DocumentationGenerator.DocumentClass(TypeInformation typeData) in C:\Dev\mddox\src\DocumentationGenerator.cs:line 306
   at MdDox.DocumentationGenerator.DocumentTypes() in C:\Dev\mddox\src\DocumentationGenerator.cs:line 349
   at MdDox.DocumentationGenerator.GenerateMarkdown(Type rootType, Assembly assembly, Boolean recursiveAssemblyTraversal, List`1 recursiveAssemblies, List`1 ignoreAttributes, Boolean ignoreMethods, IMarkdownWriter markdownWriter, String outputFileName) in C:\Dev\mddox\src\DocumentationGenerator.cs:line 127
   at MdDox.Program.Main(String[] args) in C:\Dev\mddox\src\Program.cs:line 193
```

## More information

Running just `mddox`, which displays the usage, shows I am on version 0.2.0.0:

```
T:\> mddox
mddox, version 0.2.0.0, (c) 2019 loxsmoke
Markdown documentation generator. See https://github.com/loxsmoke/mddox for more info.

Usage:
   mddox <assembly> <optional-parameters>
   <assembly>             The name of the assembly to document.

Optional parameters:
   --output <output_md>   The name of the markdown output file.
   --format <format>      The markdown file format. Valid values: github,bitbucket.
   --recursive            Step into referenced assemblies recursively.
   --recursive <assembly> Step recursivelly only into specified assembly or assemblies.
   --ignore-methods       Do not generate documentation for methods and constructors.
   --ignore-attribute <name>
                          Do not generate documentation for properties with specified custom attribute.
                          For example JsonIgnoreAttribute
   --type <name>          Document specified only this and referenced types.
```

**Comment by loxsmoke** [comment]
Generator chokes on compiler-generated generic classes that contain no backtick. Such types also do not have the CompilerGeneratedAttribute which is odd. 

I will look into this problem.

**Comment by loxsmoke** [comment]
Fixed in mddox 0.3.0. 
Had to update docxml nuget to version 3.2.2

**Comment by atifaziz** [comment]
Thank you for the quick fix!
